KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -92,6 +93,10 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     // MARK: - Private methods
 
     private func requestExternalAuthentication() {
+        guard isExternalAuthRequestInProgress == false else {
+            return
+        }
+        isExternalAuthRequestInProgress = true
         scheduleExtenalAuthInvalidationTimer()
         Karhoo.configuration.requireSDKAuthentication { [weak self] in
             guard let self = self else {
```

Comment:
missing `isExternalAuthRequestInProgress = false` inside else section. WIn case guard will fail we will end up with always set to true